examples/autoqasm/5_Pulse_programming_and_dynamical_decoupling.ipynb

```diff
@@ -0,0 +1,263 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "814647c1",
+   "metadata": {},
+   "source": [
+    "First, we create the X and Y $\\pi$ pulse in terms of the native gates of Rigetti Aspen-M-3 device. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "9130f6b8",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pi = np.pi\n",
+    "\n",
+    "def x_pulse(qubit: int):\n",
+    "    # Pi pulse apply on X-axis of Bloch sphere\n",
+    "    qubit = f\"${qubit}\"\n",
+    "    rx(qubit, pi)\n",
+    "\n",
+    "def y_pulse(qubit: int):\n",
+    "    # Pi pulse apply on Y-axis of Bloch sphere\n",
+    "    qubit = f\"${qubit}\"\n",
+    "    rz(qubit, -0.5 * pi)\n",
+    "    rx(qubit, pi)\n",
+    "    rz(qubit, +0.5 * pi)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ffb0e2d1",
+   "metadata": {},
+   "source": [
+    "Each cycle of a dynamical decoupling is a sequence of X-Y-X-Y pulses, and each of the four pulses are equally distributed in a cycle. For example, if the idling duration assigned to a cycle is $8\\tau$, each pulse is assigned to the middle of each $2\\tau$ duration. The resulting pulse distribution of a dynamical decoupling cycle is shown in the figure below.\n",
+    "\n",
+    "<img src=\"https://d2908q01vomqb2.cloudfront.net/5a5b0f9b7d3f8fc84c3cef8fd8efaaa6c70d75ab/2022/11/29/coldquanta_pulse_blog_pulses-2-1024x489.png\" width=\"500\">\n",
+    "\n",
+    "The program below realizes this pattern of the dynamical decoupling sequence. The sequence is shown as a standalone program, but the same sequence can be inserted into any program that has idling qubits."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "464ee2ed",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@aq.main\n",
+    "def idle_with_dd(qubit: int, idle_duration: float, n_cycles: int = 1) -> None:\n",
+    "    dd_spacing = idle_duration / (4*n_cycles)\n",
+    "\n",
+    "    control_frame = device.frames[f\"q{qubit}_rf_frame\"]\n",
+    "    \n",
+    "    pulse.delay(control_frame, dd_spacing)\n",
+    "    for _ in aq.range(n_cycles):\n",
+    "        x_pulse(qubit)\n",
+    "        pulse.delay(control_frame, 2*dd_spacing)\n",
+    "        y_pulse(qubit)\n",
+    "        pulse.delay(control_frame, 2*dd_spacing)\n",
+    "        x_pulse(qubit)\n",
+    "        pulse.delay(control_frame, 2*dd_spacing)\n",
+    "        y_pulse(qubit)\n",
+    "        \n",
+    "    pulse.delay(control_frame, dd_spacing)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8853db5e",
+   "metadata": {},
+   "source": [
+    "The OpenQASM script of the dynamical decoupling program is printed out below. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "e117caf0",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
```

Comment:
Maybe instead of printing this, we could run it on the simulator? Does it have an appropriate noise model?

Reply:
agree, it would be great to show the example in action. But no pulse simulator is available atm